modules/ardrivo/test/Wire.cxx

```diff
@@ -0,0 +1,194 @@
+        if (tmp_buf.size() == sizeof(T)) {
+            std::memcpy(&out.emplace_back(), tmp_buf.data(), sizeof(T));
+            tmp_buf.clear();
+        }
+    }
+}
+
+void slave_worker(const void* input, void* output, size_t stop, PacketBuffers& packet_buffer, BlockingBus<std::tuple<>>& blocking_bus,
+                  std::mutex& bus_mutex) {
+    const auto* const in = static_cast<const std::byte*>(input);
+    auto* const buf = static_cast<std::byte*>(output);
+
+    std::size_t received_bytes = 0;
+    std::size_t send_bytes = 0;
+
+    do {
+        std::scoped_lock tx_lock{bus_mutex, packet_buffer.tx_mutex};
+        if (!packet_buffer.tx.empty()) {
+            std::copy(packet_buffer.tx.front().data(), packet_buffer.tx.front().data() + packet_buffer.tx.front().size(),
+                      buf + received_bytes);
+            received_bytes += packet_buffer.tx.front().size();
+            packet_buffer.tx.erase(packet_buffer.tx.begin());
+        }
+        std::scoped_lock rx_lock{packet_buffer.rx_mutex};
+        if (blocking_bus.request_bytes > 0) {
+            auto& ret = packet_buffer.rx.emplace_back();
+            ret.resize(blocking_bus.request_bytes);
+            std::copy(in + send_bytes, in + send_bytes + blocking_bus.request_bytes, ret.begin());
+            send_bytes += blocking_bus.request_bytes;
+            blocking_bus.request_bytes = 0;
+            blocking_bus.sync.notify_all();
+        }
+    } while (send_bytes < stop);
+}
+
+void transfer_test(std::uint8_t bus_no, std:: uint8_t slave_no){
+    const auto chunk_size = GENERATE(take(256 / sizeof(unsigned), random(1, 8)));
+    const std::vector slave_in = GENERATE_COPY(
+        take(1, chunk(chunk_size, random<unsigned>(std::numeric_limits<std::uint8_t>::min(), std::numeric_limits<std::uint8_t>::max()))));
+    std::vector<decltype(slave_in)::value_type> slave_out(slave_in.size());
+    const std::vector master_in = GENERATE_COPY(
+        take(1, chunk(chunk_size, random<unsigned>(std::numeric_limits<std::uint8_t>::min(), std::numeric_limits<std::uint8_t>::max()))));
+    std::vector<decltype(slave_in)::value_type> master_out{};
+
+    auto& blocking_bus = std::get<BlockingBus<std::tuple<>>>(board_data->i2c_buses[bus_no].slaves[slave_no].second);
+    auto& packet_buffer = board_data->i2c_buses[bus_no].slaves[slave_no].first;
+    auto& bus_mut = board_data->i2c_buses[bus_no].devices_mut;
+    board_data->i2c_buses[bus_no].slaves[slave_no].second.emplace<0>();
+    auto slave_thread = std::thread(slave_worker, slave_in.data(), slave_out.data(), (size_t)slave_in.size() * sizeof(unsigned), std::ref(packet_buffer), std::ref(blocking_bus), std::ref(bus_mut));
+
+    Wire.begin(bus_no, bus_no);
+    Wire.beginTransmission(slave_no);
+    Wire.write(reinterpret_cast<const std::uint8_t*>(master_in.data()), master_in.size() * sizeof(unsigned));
+    Wire.endTransmission();
+
+    Wire.requestFrom(slave_no, (size_t)slave_in.size() * sizeof(unsigned), false);
```

Comment:
Identity C-style cast; discard

Reply:
If you're compiling the tests with GCC then you could add `-Wold-style-cast` as a compiler flag.

Reply:
We do allow C-style widening integral casts in the codebase; however this is not widening since it is an identity cast
You do raise the issue of a proper flags setup so that people other than me actually go dig the warnings

Reply:
[For codebase contribs:] Only explicitly widen when function call resolution or unary `operator+` are not sufficient